feature_engine/variable_manipulation.py

```diff
@@ -221,14 +221,23 @@ def _find_or_check_datetime_variables(
             raise ValueError("No datetime variables found in this dataframe.")
 
     elif isinstance(variables, (str, int)):
+        if variables == "index":
```

Comment:
I would not modify this function. I would handle the logic from the datetime class, because this is probably one of the only classes that will work with the index in this way.